include/boost/geometry/strategies/cartesian/intersection.hpp

```diff
@@ -68,6 +68,57 @@ namespace boost { namespace geometry
 namespace strategy { namespace intersection
 {
 
+namespace detail_usage
+{
+
+// When calculating the intersection, the information of "a" or "b" can be used.
+// Theoretically this gives equal results, but due to floating point precision
+// there might be tiny differences. These are edge cases.
+// This structure is to determine if "a" or "b" should be used.
+// Prefer the segment closer to the endpoint.
+// If both are about equally close, then prefer the longer segment
+// To avoid hard thresholds, behavior is made fluent.
+// Calculate comparable length indications,
+// the longer the segment (relatively), the lower the value
+// such that the shorter lengths are evaluated higher and will
+// be preferred.
+template <bool IsArithmetic>
+struct use_a
+{
+  template <typename Ct, typename Ev>
+  static bool apply(Ct cla, Ct clb, Ev eva, Ev evb)
+  {
+      auto const clm = (std::max)(cla, clb);
+      if (clm <= 0)
+      {
+          return true;
+      }
+
+      // Relative comparible length
+      auto const rcla = 1.0 - cla / clm;
+      auto const rclb = 1.0 - clb / clm;
+
+      // Multipliers for edgevalue (ev)
+      // and relative comparible length (rcl)
+      using ctype = decltype(eva);
+      constexpr ctype mev = 5.0;
+      constexpr ctype mrcl = 1.0;
+      return mev * eva + mrcl * rcla > mev * evb + mrcl * rclb;
+  }
+};
+
+// Specialization for non arithmetic types. They will always use "a"
+template <>
+struct use_a<false>
+{
+    template <typename Ct, typename Ev>
+    static bool apply(Ct , Ct , Ev , Ev )
```

Comment:
`const&` in case these are user-defined types.

Reply:
:heavy_check_mark: